Add custom ESLint rule to prevent importing from reactviews directory in extension.

aasimkhan30 opened the pull request:
Importing from the reactviews directory can break our VSIX packages, as it may introduce dependencies that aren't compatible with the Node environment. These issues typically don't surface during compile time, making them especially problematic. Errors only appear at runtime when activating the extension, after installing the VSIX.

![image](https://github.com/user-attachments/assets/8d63d1c1-6f40-410d-bfbd-e27100831ed1)

After using this custom rule, I will get lint errors and I cannot even push stuff in main because my commit hooks will prevent me from doing that. 

![image](https://github.com/user-attachments/assets/65f87f17-845d-4220-9917-4d70fefc4d80)

caohai commented:
Looks like we already have a bunch of "violations" across features, do we think it's necessary to fix all of them?

aasimkhan30 commented:
Hey @caohai, I just fixed the issue. 
This bug is bad as it only affects VSIX builds and doesn’t repro in dev environments. So it can be a while before we catch it. 
I moved everything to shared interfaces to keep things consistent across the codebase.
